library/core/src/fmt/mod.rs

```diff
@@ -332,6 +332,15 @@ enum FlagV1 {
 }
 
 impl<'a> Arguments<'a> {
+    /// When using the format_args!() macro, this function is used to generate the
+    /// Arguments structure in case it's just a single string literal.
+    #[doc(hidden)]
+    #[inline]
+    #[unstable(feature = "fmt_internals", reason = "internal to format_args!", issue = "none")]
```

Comment:
Is it possible to make it const-unstable at the same time under a different feature gate?

Reply:
It is const-unstable (implicitly). But the `format_args!` macro expanded code has some kind of span that still lets stable code call this.

Reply:
Only the `fmt_internals` feature is allowed:

https://github.com/rust-lang/rust/blob/a84d1b21aea9863f0fc5f436b4982d145dade646/library/core/src/macros/mod.rs#L826

Using a different feature for const checking would require this feature to be enabled by the user of `format_args!`.

Reply:
The reason i want this is so `panic!()` can use this for const panic. (See https://github.com/rust-lang/rust/pull/86998) Putting an `allow_internal_unstable(const_fmt_internals)` on the `panic` macro won't work through the nested format_args call.

Reply:
That would at least prevent it from being insta-stable, right? Maybe use `const_panic` as feature gate?

Reply:
I think we should just go for the `const_format_args!()` macro for now (#86998) and at some point stabilize `format_args!()` in const entirely. (And then remove `const_format_args!()` again.)